docs/topics/jvm/jvm-spring-boot-add-db-support.md

```diff
@@ -0,0 +1,328 @@
+[//]: # (title: Add database support for Spring Boot project)
+[//]: # (description: Add a database support for Sprint Boot project written in Kotlin using JDBC template.)
+
+<microformat>
+    <p>This is the third part of the <strong>Getting started with Spring Boot and Kotlin</strong> tutorial. Before proceeding, make sure you've completed previous steps:</p><br/>
+    <p><img src="icon-1-done.svg" width="20" alt="First step"/> <a href="jvm-create-project-with-spring-boot.md">Create a Spring Boot project with Kotlin</a><br/><img src="icon-2-done.svg" width="20" alt="Second step"/> <a href="jvm-spring-boot-add-data-class.md">Add a data class to the Spring Boot project</a><br/><img src="icon-3.svg" width="20" alt="Third step"/> <strong>Add database support for Spring Boot project</strong><br/><img src="icon-4-todo.svg" width="20" alt="Fourth step"/> Use Spring Data CrudRepository for database access</p>
+</microformat>
+
+In this part of the tutorial you'll add and configure a database to your project using JDBC. In JVM applications, you use JDBC to interact with databases.
+For convenience, Spring Framework provides `JdbcTemplate` class that simplifies the use of JDBC and helps to avoid common errors.
+
+## Add database support
+
+The common practice in Spring Framework based applications is to implement the database access logic within the so-called _service_ layer – this is where business logic lives.
+In Spring, you should mark classes with the `@Service` annotation to imply that the class belongs to the service layer of the application.
+In this application, you will create the `MessageService` class for this purpose.
+
+In `DemoApplication.kt` file, create `MessageService` class as follows:
+
+```kotlin
+import org.springframework.stereotype.Service
+import org.springframework.jdbc.core.JdbcTemplate
+
+@Service
+class MessageService(val db: JdbcTemplate) {
+    fun findMessages(): List<Message> = db.query("select * from messages") { rs, _ ->
+        Message(rs.getString("id"), rs.getString("text"))
```

Comment:
```suggestion
        Message(response.getString("id"), response.getString("text"))
```

Reply:
I would recommend not using abbreviations for non-standard variables' names. `db` is quite standard and `rs` is not. I suppose we would like to encourage our users to write clean code.

If you decide to accept this edit, need to edit such cases below too.